### PR TITLE
Add the device authorization endpoint

### PR DESCRIPTION
**Pull request by oliamb**
This is a follow-up PR proposal for issue #626. It implements Section 3.1 and Section 3.2 of [RFC8628](https://www.rfc-editor.org/rfc/rfc8628#section-3.1). PR #640 is related and arguably more advanced. But it is a bit old now and relates to the draft specification.

This is my first submission here. I think I followed the Contributing Guidelines, but please let me know if I am doing it wrong.

### CODE REVIEW

**Comment by dopry** [comment]
@oliamb do you have time to address the lint issues, that might help get this on a maintainers radar for review.

**Comment by oliamb** [comment]
Thanks @dopry. I did not even see that there was a lint issue. This should be fixed now.

**Comment by auvipy** [comment]
lets wait for the CI

**Comment by oliamb** [comment]
The linter fails at other points now, but I can see this is from code outside the original PR. I rebased my branch on top of the master yesterday. I don't think I should fix files outside the PR scope, WDYT? I don't manage to get consistent results on my local with the linter because I think the linter libraries' versions are not pinned (i.e. they are installed through the command `pip install black codespell mypy pytest ruff safety`).

**Comment by dopry** [comment]
I would treat upstream lint issues as out of scope, if I did want to fix them I would submit it as a separate PR.

**Comment by dopry** [comment]
@auvipy do you have any thoughts on clearing the lint issues?

**Review by thedrow — APPROVED** [review]

File: tests/oauth2/rfc8628/test_server.py
Comment (auvipy) [inline tests/oauth2/rfc8628/test_server.py:1]:
```suggestion
```

**Comment by jasperbok** [comment]
I could use the changes in this PR, but I see it cannot be merged due to linting errors. As https://github.com/oauthlib/oauthlib/pull/871 got merged earlier this year, I thought perhaps rebasing this branch on `master` would help make the linting pass, but unfortunately it didn't. Is there anything I can do to turn this PR into a success? I could make a separate PR to resolve the linting issues if that's OK?

**Comment by thedrow** [comment]
@jasperbok Please do fix the linter issue.

**Comment by duzumaki** [comment]
I'm hoping [this](https://github.com/oauthlib/oauthlib/pull/881) fixes the linter issue

**Comment by duzumaki** [comment]
I'm hoping [this](https://github.com/oauthlib/oauthlib/pull/881) fixes the linter issue @thedrow

**Comment by thedrow** [comment]
Thank you guys for your contributions 🙏 .

**Pull request closed without merging**
